### PR TITLE
Update torch to 1.13.1 in CI and packaging pipelines for ort training

### PR DESCRIPTION
**Pull request by baijumeswani**
Addressing [CG issue](https://aiinfra.visualstudio.com/Lotus/_componentGovernance/97926/alert/8088472)

Also, since ORTTrainer is deprecated, this PR disables the ORTTrainer tests in the CI pipeline.

### CODE REVIEW

**Review by adrianlizarraga — COMMENTED** [review]
Hi Baiju,

Thanks for addressing this issue. Here are a few other locations that may need a torch version update. I did not see them in the files changed.

- tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/torch_eager_cpu/requirements.txt
- tools/ci_build/github/linux/docker/scripts/training/requirements.txt
- tools/ci_build/github/windows/eager/requirements.txt

**Comment by baijumeswani** [comment]
Hi Adrian, from the CG issue, I found that the only impacted component was torch 1.12.1:
image
Can you confirm if torch 1.12.0 is also an impacted component?

**Comment by adrianlizarraga** [comment]
> Can you confirm if torch 1.12.0 is also an impacted component?

Yes, versions <= 1.13.0 are affected: https://github.com/advisories/GHSA-47fc-vmwq-366v

torch_cve

**Review by snnn — APPROVED** [review]

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -1942,11 +1780,6 @@ def run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs):
             if not args.disable_ml_ops and not args.use_tensorrt:
                 run_subprocess([sys.executable, "onnxruntime_test_python_mlops.py"], cwd=cwd, dll_path=dll_path)
 
-            # The following test has multiple failures on Windows
-            if args.enable_training and args.use_cuda and not is_windows():
-                # run basic frontend tests
-                run_training_python_frontend_tests(cwd=cwd)
```
Comment (adrianlizarraga) [inline tools/ci_build/build.py:1948]: If these are going to be reenabled in the future, would it make sense to add a TODO or log with an explanation?

```python
if args.enable_training and args.use_cuda and not is_windows():
    # TODO: Add tests when ....
   log.info("...")
```

Comment (baijumeswani) [reply tools/ci_build/build.py:1948]: These are not going to be re-enabled in the future. These tests are part of the older `ORTTrainer` code. This has been deprecated and removing the unit tests was part of the plan for a while.
There was only one test here that needs to be re-enabled, and I have moved it to https://github.com/microsoft/onnxruntime/pull/14055/files#diff-10e9937810323c48f9c951ef10a99ba7f16829893232d59584242499eccbc76fR189-R191

**Review by adrianlizarraga — APPROVED** [review]

**Comment by baijumeswani** [comment]
@microsoft/onnxruntime-es team, please review when possible.

**Review by jchen351 — DISMISSED** [review]

**Review by jchen351 — APPROVED** [review]

**Pull request merged**